go/service/main.go

```diff
@@ -102,16 +108,21 @@ func (d *Service) Handle(c net.Conn) {
 		logReg = newLogRegister(d.logForwarder, d.G().Log)
 		defer logReg.UnregisterLogger()
 	}
-
-	if err := d.RegisterProtocols(server, xp, connID, logReg, d.G()); err != nil {
+	shutdowners, err := d.RegisterProtocols(server, xp, connID, logReg, d.G())
+	defer func() {
```

Comment:
Could you elaborate a bit more?

If you mean if the function is guaranteed to return, the channel returned by `server.Run()` below at line 123 is closed here: https://github.com/keybase/go-framed-msgpack-rpc/blob/ffba94f73657451b549a28cc779f55cccc72c032/transport.go#L159 which is triggered on "transport error"